Mathlib/Algebra/GroupWithZero/NonZeroDivisors.lean

```diff
@@ -277,3 +275,89 @@ lemma nonZeroSMulDivisors_mulOpposite_eq_op_nonZeroDivisors :
     Rᵐᵒᵖ ⁰[R] = R⁰.op := rfl
 
 end nonZeroSMulDivisors
+
+noncomputable section Associated
+
+open nonZeroDivisors
+
+section MonoidWithZero
+
+variable (α : Type*) [MonoidWithZero α]
+
+/-- The units of the monoid of non-zero divisors of `α` are equivalent to the units of `α`. -/
+def nonZeroDivisorsUnitsEquiv : α⁰ˣ ≃* αˣ :=
+  MulEquiv.ofBijective (Units.map α⁰.subtype) ⟨Units.map_injective Subtype.val_injective,
+    fun u ↦ ⟨IsUnit.unit ⟨⟨⟨u, IsUnit.mem_nonZeroDivisors u.isUnit⟩, ⟨(u⁻¹ : αˣ),
+        IsUnit.mem_nonZeroDivisors u⁻¹.isUnit⟩, by simp, by simp⟩, rfl⟩,
+      by rw [Units.ext_iff, IsUnit.unit_of_val_units, Units.coe_map, Submonoid.coe_subtype]⟩⟩
```

Comment:
This equivalence can be given fully computably, right? Mind making it so, so that `nonZeroDivisorsUnitsEquiv_symm_apply` is defeq?

Reply:
`MulEquiv.ofBijective` is noncomputable so I would need to use something else.

Reply:
I'll try using `MulHom.toMulEquiv` instead.

Reply:
Hum, `IsUnit.unit` is also noncomputable...

Reply:
Any way it works